src/OpenTelemetry/Metrics/MetricStreamConfiguration.cs

```diff
@@ -91,11 +95,36 @@ public class MetricStreamConfiguration
         }
     }
 
+    /// <summary>
+    /// Gets or sets a positive integer value
+    /// defining the maximum number of data points allowed to for the metric
+    /// managed by the view.
+    /// </summary>
+    /// <remarks>
+    /// Note: If not set, the SDK cardinality limit value will be used, which
+    /// defaults to 2000. Call <see cref="MeterProviderBuilderExtensions"/>
+    /// SetMaxMetricPointsPerMetricStream"/> to confiture the SDK defaults.
+    /// </remarks>
+#if EXPOSE_EXPERIMENTAL_FEATURES
+    public
+#else
+    internal
+#endif
+    int? CardinalityLimit
+    {
+        get => this.cardinalityLimit;
+        set
+        {
+            if (value != null)
+            {
+                Guard.ThrowIfOutOfRange(value.Value, min: 3, max: int.MaxValue);
```

Comment:
@Yun-Ting would you put a comment here explaining why `min` is 3? (consider updating the XML doc to call out the accepted range)

Reply:
Yeah, one for the overflowAttribute when `emitOverflowAttribute` is set to true, and the other for 0 tag, assuming when users want set this value, they _want_ to passed in tags.

Currently, the lower limit for `.SetMaxMetricPointsPerMetricStream()` is [1](https://github.com/open-telemetry/opentelemetry-dotnet/blob/a31bca866dab2f9735713e9be9bcf80d95ca0104/src/OpenTelemetry/Metrics/Builder/MeterProviderBuilderExtensions.cs#L215C1-L215C59), though.

In [AggregatorStore](https://github.com/open-telemetry/opentelemetry-dotnet/blob/a31bca866dab2f9735713e9be9bcf80d95ca0104/src/OpenTelemetry/Metrics/AggregatorStore.cs#L96C9-L102C10), we reserved one for the overflowAttribute.

And in doc, it says one should be reserved for [0 tag](https://github.com/open-telemetry/opentelemetry-dotnet/tree/a31bca866dab2f9735713e9be9bcf80d95ca0104/docs/metrics/customizing-the-sdk#:~:text=One%20MetricPoint%20is%20reserved%20for%20every%20MetricStream%20for%20the%20special%20case%20where%20there%20is%20no%20key/value%20pair%20associated%20with%20the%20metric.%20The%20maximum%20number%20of%20MetricPoints%20has%20to%20accommodate%20for%20this%20special%20case.) (no key/value pair associated).

Also, there had been discussions on whether we should reserve 1 for zero dimension: https://github.com/open-telemetry/opentelemetry-dotnet/pull/2635#discussion_r752490441.

Maybe this topic indeed worth another PR.